### ISSUE TITLE
Support for "ATI2" fourcc format (BC5)

### ISSUE THREAD

**Issue by mellinoe** [nickbabcock/Pfim#23]
Many of the textures in [this free asset pack](https://developer.nvidia.com/orca/amazon-lumberyard-bistro) are in BC5 format. Trying to load the file yields the following:

```
System.AggregateException: 'One or more errors occurred. (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.) (FourCC: 843666497 not supported.)'
```

I'm mainly / only interested in getting at the raw compressed data (and the header info), so if decoding the pixel data is prohibitively complex, then it would be fine if the compressed data was still accessible.

**Comment by nickbabcock** [comment]
Ok yeah let me take a look. One thing I noticed is that the "Antenna_Metal_0_Normal.dds" has a size of 5.333MiB on disk yet calculating how bytes the image should take up (2048 x 2048 with 16 bytes per 4 x 4 pixel block) = 4MiB. I need to look a bit more into it.

**Comment by mellinoe** [comment]
The images all have mipmaps, so there is more data in the file than just the first mipmap level. E.g. that 2048x2048 image has 11 mipmap levels.

**Comment by nickbabcock** [comment]
Oh yes that is right. Is it important that the mipmaps are exposed (a TODO)? Else testing locally, the ATI2 decoder worked on the several images tested from the asset pack.

**Comment by mellinoe** [comment]
Yes, getting at all of the mipmaps would be needed for completeness. Again, I just need the raw bytes though, no processing or anything necessary.

**Comment by nickbabcock** [comment]
Ok sounds good, I have an impl that will preserve mipmap data when choosing to skip processing (the mipmap data isn't processed either). This should be what you're looking for, but I'll mull over it until potentially tomorrow.

Mipmap data will be lost if the image is subsequently processed (`IImage::Decompress`).

**Comment by nickbabcock** [comment]
Changes committed. If no issues arise, I'll cut a release tonight

**Comment by nickbabcock** [comment]
Just released 0.5.2
